common/beerocks/bwl/shared/nl80211_client_impl.cpp

```diff
@@ -17,13 +21,276 @@
+                        uint16_t max_bandwidth_MHz = 0;
+                        static struct nla_policy freq_policy[NL80211_FREQUENCY_ATTR_MAX + 1];
+                        freq_policy[NL80211_FREQUENCY_ATTR_FREQ]         = {NLA_U32, 0, 0};
+                        freq_policy[NL80211_FREQUENCY_ATTR_DISABLED]     = {NLA_FLAG, 0, 0};
+                        freq_policy[NL80211_FREQUENCY_ATTR_RADAR]        = {NLA_FLAG, 0, 0};
+                        freq_policy[NL80211_FREQUENCY_ATTR_MAX_TX_POWER] = {NLA_U32, 0, 0};
+
+                        nla_for_each_nested(nl_freq, tb_band[NL80211_BAND_ATTR_FREQS], rem_freq)
+                        {
+                            struct nlattr *tb_freq[NL80211_FREQUENCY_ATTR_MAX + 1];
+                            sChannelInfo channel{};
+
+                            if (nla_parse(tb_freq, NL80211_FREQUENCY_ATTR_MAX,
+                                          static_cast<nlattr *>(nla_data(nl_freq)),
+                                          nla_len(nl_freq), freq_policy)) {
+                                LOG(ERROR)
+                                    << "Failed to parse frequency " << nl_freq->nla_type + 1 << "!";
+                                return false;
+                            }
+
+                            if (!tb_freq[NL80211_FREQUENCY_ATTR_FREQ] ||
+                                tb_freq[NL80211_FREQUENCY_ATTR_DISABLED] ||
+                                (tb_freq[NL80211_FREQUENCY_ATTR_NO_IR] &&
+                                 tb_freq[__NL80211_FREQUENCY_ATTR_NO_IBSS]) ||
+                                tb_freq[__NL80211_FREQUENCY_ATTR_NO_IBSS]) {
+                                continue;
+                            }
+
+                            if (tb_band[NL80211_FREQUENCY_ATTR_FREQ]) {
+                                uint32_t freq   = nla_get_u32(tb_freq[NL80211_FREQUENCY_ATTR_FREQ]);
+                                channel.channel = ieee80211_frequency_to_channel(freq);
+                            } else {
+                                LOG(DEBUG) << "NL80211_FREQUENCY_ATTR_FREQ attribute is missing";
+                            }
+
+                            if (tb_freq[NL80211_FREQUENCY_ATTR_MAX_TX_POWER]) {
+                                channel.tx_pow_dBm =
+                                    0.01 *
+                                    nla_get_u32(tb_freq[NL80211_FREQUENCY_ATTR_MAX_TX_POWER]);
+                            } else {
+                                LOG(DEBUG)
+                                    << "NL80211_FREQUENCY_ATTR_MAX_TX_POWER attribute is missing";
+                            }
+
+                            if (tb_freq[NL80211_FREQUENCY_ATTR_RADAR]) {
+                                channel.is_dfs = true;
+                            }
+
+                            if (!tb_freq[NL80211_FREQUENCY_ATTR_NO_20MHZ]) {
+                                channel.bandwidth_MHz = 20;
+                            }
```

Comment:
This will only be set for channels that are limited to 5 or 10MHz, which is probably not something we will ever want to use. So I think we can safely skip a channel that has NO_20MHZ set.

If not, we'll have to initialize bandwidth_MHz with a non-zero value, or it will *sometimes* be ignored...

Reply:
Done